developer/utils.py

```diff
@@ -0,0 +1,166 @@
+import os
+import optparse
+
+
+from tank import LogManager
+from tank.util import filesystem
+from tank.platform import environment
+from tank.descriptor import Descriptor, create_descriptor
+from tank.authentication import ShotgunAuthenticator
+
+logger = LogManager.get_logger("utils")
+
+
+class OptionParserLineBreakingEpilog(optparse.OptionParser):
```

Comment:
rather than having a catch-all utils file, could we split this up into several files for clarity? Maybe one for the option parser stuff, one for the app caching stuff?